### PR TITLE
chore(risedev): bump cargo-make version

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

For example, changes introduced in #9006 use a recent functionality `files_modified" of `cargo-make`.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
Actually, it's added in a patch version of `cargo-make` (0.36.6) 😅, so I'm also wishing to compare the patch version, that is, whether the version meets the requirement of "~0.36.6". Though `cargo install` can already handle updates after https://github.com/rust-lang/cargo/issues/6667, it requires the network connection to work. Any ideas from my dear @xxchan?

File: risedev [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 
-if [ -z "$(which cargo-make)" ]; then
+if [[ $(cargo make --version) != "cargo-make 0.36"* ]]; then
```
Comment (xxchan) [inline risedev:3]: >= ?

Comment (TennyZhuang) [reply risedev:3]: It's hard to do ">" for semver in bash.

**Review by TennyZhuang — APPROVED** [review]
LGTM

File: risedev (hunk 1 above)
Comment (xxchan) [reply risedev:3]: 👼

File: risedev [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
     echo "Installing cargo-make..."
-    cargo install cargo-make --version "^0.35" --locked
+    cargo install cargo-make --version "^0.36" --locked
```
Comment (xxchan) [inline risedev:5]: Should it be
```suggestion
    cargo install cargo-make --version "~0.36" --locked
```
?

**Comment by xxchan** [comment]
My honey @BugenZhao, I think it's good enough, although occasionally breaks 😇. If you really want to fix it, I think we should also consider how to manage versions for e.g., `cargo-hakari`.

**Comment by xxchan** [comment]
BTW, cargo make supports installing dependencies and specifying versions https://github.com/sagiegurari/cargo-make#defining-version 🤔

**Comment by xxchan** [comment]
Use 
```
[config]
min_version = "0.36.6"
```
instaed

**Comment by BugenZhao** [comment]
> sagiegurari/cargo-make@`d24e1d1`#minimal-version

This is really cool. 🫡 Thank you, my sweet @xxchan.

**Pull request merged**